### PR TITLE
fix Python 3 regressions

### PR DESCRIPTION
**Pull request by dirk-thomas**
Regressions from #63.

@eric-wieser FYI

### CODE REVIEW

File: src/genpy/rostime.py [hunk 1]
```diff
@@ -226,6 +228,9 @@ def to_time(self):
         """
         return self.to_sec()
 
+    def __hash__(self):
+        return super(Time, self).__hash__()
```
Comment (eric-wieser) [inline src/genpy/rostime.py]: Why are these needed? Doesn't the base class provide them>

File: src/genpy/rostime.py [hunk 2]
```diff
@@ -180,7 +180,9 @@ def __ne__(self, other):
     def __cmp__(self, other):
```
Comment (eric-wieser) [inline src/genpy/rostime.py]: Arguably the real issue here is implementing `__cmp__` in python 3 at all. If `cmp` doesn't exist, then `__cmp__` will never be called. So I don't understand what this patch is fixing.

Comment (eric-wieser) [inline src/genpy/rostime.py]: Ah, I see what the issue is - the implementation of `__lt__` delegates to `__cmp__`

File: src/genpy/rostime.py (hunk 1 above)
Comment (eric-wieser) [inline src/genpy/rostime.py]: Sorry, clearly the base class does provide them, because we're calling `super`!

What I meant to say, is why do we need to overload them? This works just fine for me:

```python
#! python3.5

class Base:
    def __hash__(self):
        return 4

class Derived(Base): pass

class DerivedWithOverride(Base):
    def hash(self):
        return super(DerivedWithOverride, self).__hash__()

assert hash(Base()) == hash(DerivedWithOverride()) == hash(Derived())

Comment (eric-wieser) [inline src/genpy/rostime.py]: Can you give me the full traceback?

Comment (dirk-thomas) [inline src/genpy/rostime.py]:
```
Traceback (most recent call last):
  File "/home/dthomas/ros/kinetic/github/ros/genpy/test/test_genpy_rostime.py", line 180, in test_Time
    self.test_TVal(TVal=Time, test_neg=False)
  File "/home/dthomas/ros/kinetic/github/ros/genpy/test/test_genpy_rostime.py", line 61, in test_TVal
    self.assertEquals(v.__hash__(), TVal(0, 0).__hash__())
TypeError: 'NoneType' object is not callable
```

Comment (eric-wieser) [inline src/genpy/rostime.py]: Ok, I see it now. Something weird seems to be happening, so in the absence of knowing the reason, reintroducing these lines makes sense!

**Comment by eric-wieser** [comment]
I've got a DNS outage right now, so I'm assuming that was "No, this is
needed", as i forget the phrasing of my comment.

In python 2 or 3? This works just fine on my machine running python 3.5:

>>> class Foo:
def __hash__(self):
return 3

>>> class Bar(Foo): pass

>>> hash(Foo())
3
>>> hash(Bar())
3
